pkg/urbit/vere/daemon.c

```diff
@@ -612,7 +612,13 @@ _boothack_key(u3_noun kef)
 
     //  +seed:able:jael: private key file
     //
-    seed = u3ke_cue(u3k(u3t(des)));
+    seed = u3m_soft(0, u3ke_cue, u3k(u3t(des)));
+    if ( u3_blip != u3h(seed) ) {
+      u3l_log("dawn: unable to cue private key\r\n");
+      exit(1);
+    }
+    seed = u3t(seed);
```

Comment:
This logic looks right, but it's leaking the reference to the product of `u3m_soft()`. You need to gain a reference to the tail of the product of `u3m_soft()`, and lose a reference to the product itself. This is tedious, and tends to require lots of local variables (which has its own implications for code style ....).

Something like this:

```
{
  u3_noun pro = u3m_soft(...);

  if ( u3_blip != u3h(pro) ) {
    // fatal error, refcounts ignored
  }
  else {
    foo = u3k(u3t(pro));
    u3z(pro);
  }
}
```

Reply:
thanks -- reassigning to `seed` was probably bad style anyway. Think I got it now.